src/views/collections/CreateCollection/EditCollection.vue

```diff
@@ -0,0 +1,234 @@
+            </b-button>
+            <b-modal id="delete" title="Delete Collection?" hide-footer>
+              <div class="d-block text-center">
+                Warning: This will permanently delete the collection!
+              </div>
+
+              <b-button
+                class="mt-3"
+                type="submit"
+                style="margin-left: 10px"
+                variant="primary"
+                @click="$bvModal.hide('delete')"
+              >
+                {{ 'Cancel' }}
+              </b-button>
+              <b-button
+                class="mt-3"
+                type="submit"
+                style="margin-left: 10px"
+                variant="danger"
+                @click="deleteCollection()"
+              >
+                {{ 'Delete' }}
+              </b-button>
+            </b-modal>
+            <b-button
+              type="submit"
+              style="margin-left: 10px"
+              variant="danger"
+              v-b-modal.delete
+            >
+              {{ 'Delete' }}
+            </b-button>
+          </div>
+        </div>
+      </div>
+    </div>
+  </EternaPage>
+</template>
+
+<script lang="ts">
+// @ts-ignore
+  import debounce from 'lodash.debounce';
+  import {
+    Component,
+    Vue,
+    Mixins,
+    Watch,
+    Ref,
+    Prop,
+  } from 'vue-property-decorator';
+  import { RouteCallback, Route } from 'vue-router';
+  import axios, { AxiosInstance } from 'axios';
+  // @ts-ignore
+  import EternaPage from '@/components/PageLayout/EternaPage.vue';
+  import TagsPanel from '@/components/Sidebar/TagsPanel.vue';
+  import Utils from '@/utils/utils';
+  import { CollectionItem, PuzzleItem } from '@/types/common-types';
+  import FetchMixin from '@/mixins/FetchMixin';
+  import CollectionInfo from './components/CollectionInfo.vue';
+  import CollectionPuzzles from './components/CollectionPuzzles.vue';
+
+  @Component({
+    components: {
+      EternaPage,
+      TagsPanel,
+      CollectionInfo,
+      CollectionPuzzles,
+    },
+  })
+  export default class EditCollection extends Mixins(FetchMixin) {
+    @Prop({ required: true }) private loading!: boolean;
+
+    private title = '';
+
+    private desc = '';
+
+    private puzzlenames: PuzzleItem[] = [];
+
+    private puzzlelist: PuzzleItem[] = [];
+
+    private picture: string = '';
+
+    private newPicture: File | null = null;
+
+    private data: boolean = false;
+
+    fetchData: () => Promise<void> | undefined = async () => {};
+
+    async fetch() {
+      const collection = (
+        await this.$http.get(`/get/?type=collection&nid=${this.$route.params.id}`)
+      ).data.data.collection as CollectionItem;
+      if (collection.puzzles != null) {
+        const puzzlelist = collection.puzzles.split(',');
+        const res = await Promise.all(
+          Object.values(puzzlelist).map((puzz) =>
+            this.$http.get(`/get/?type=puzzle&nid=${parseInt(puzz, 10)}`),
```

Comment:
You shouldn't need to parseInt, since you're just putting it right back into a string